include/ofi_util.h

```diff
@@ -267,6 +267,83 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 
 int ofi_endpoint_close(struct util_ep *util_ep);
 
+/*
+ * Receive Queue functionality with tag matching and directed recv support
+ */
```

Comment:
> I'm guessing @aingerson would have loved this about 5 days ago..

My time machine order didn't go through unfortunately. ;)

I think this code can be used by RxM provider since the tag matching functions are very similar. I had RxM and multi-rail providers in mind when trying to pull out re-usable parts of tag matching. I believe it should be usable by tcp and shm.

Reply:
I spoke with Alexia.  The rxd code was merged, and she'll look at moving both rxd and shm to using whatever utility code gets merged.